azurelinuxagent/common/cgroups.py

```diff
@@ -0,0 +1,689 @@
+            Add this process to the "agent" cgroup, if required
+        Actual collection of metrics from cgroups happens in the -run-exthandlers instance
+        """
+        cgroups_enabled = False
+        try:
+            CGroups._osutil.mount_cgroups()
+            if not suppress_process_add:
+                CGroups._setup_wrapper_groups()
+                pid = int(os.getpid())
+                if not CGroups.is_systemd_manager():
+                    cg = CGroups.for_extension(AGENT_NAME)
+                    cg.add(pid)
+                    logger.info("Add daemon process pid {0} to {1} cgroup".format(pid, cg.name))
+                else:
+                    logger.info("Daemon process pid {0} cgroup managed by systemd".format(pid))
+            cgroups_enabled = True
+            status = "OK"
+        except CGroupsException as cge:
+            status = cge.msg
+
+        from azurelinuxagent.common.event import add_event, WALAEventOperation
+        add_event(
+            AGENT_NAME,
+            version=CURRENT_VERSION,
+            op=WALAEventOperation.InitializeCGroups,
+            is_success=cgroups_enabled,
+            message=status,
+            log_event=False)
+
+    @staticmethod
+    def add_to_extension_cgroup(name, pid=int(os.getpid())):
+        """
+        Create cgroup directories for this extension in each of the hierarchies and add this process to the new cgroup.
+        Should only be called when creating sub-processes and invoked inside the fork/exec window. As a result,
+        there's no point in returning the CGroups object itself; the goal is to move the child process into the
+        cgroup before the new code even starts running.
+
+        :param str name: Short name of extension, suitable for naming directories in the filesystem
+        :param int pid: Process id of extension to be added to the cgroup
+        """
+        if not CGroups.enabled():
+            return
+        if name == AGENT_NAME:
+            logger.warn('Extension cgroup name cannot match agent cgroup name ({0})'.format(AGENT_NAME))
+            return
+
+        try:
+            logger.info("Move process {0} into cgroups for extension {1}".format(pid, name))
+            CGroups.for_extension(name).add(pid)
+        except Exception as ex:
+            logger.warn("Unable to move process {0} into cgroups for extension {1}: {2}".format(pid, name, ex))
+
+    @staticmethod
+    def get_my_cgroup_path(hierarchy_id):
+        """
+        Get the cgroup path "suffix" for this process for the given hierarchy ID. The leading "/" is always stripped,
+        so the suffix is suitable for passing to os.path.join(). (If the process is in the root cgroup, an empty
+        string is returned, and os.path.join() will still do the right thing.)
+
+        :param hierarchy_id: str
+        :return: str
+        """
+        cgroup_paths = fileutil.read_file("/proc/self/cgroup")
```

Comment:
Lost comments?

I believe this is an example of what you're parsing, correct?  If so, please add in a comment.

```text
3:cpu,cpuacct:/user.slice
```

Please add an example.  Please consider encapsulating this parsing to return an object to make code read easier for a future PR.